### PR TITLE
Ice Cream Cone Tweaks and New Sprites

### PR DESCRIPTION
**Pull request by PacifistDalek**
## What Does This PR Do
Replaces the sprite for the ice cream itself, and adds a new waffle cone option to the ice cream maker, both sprites by Crumpaloo.

Tweaks a few aspects of the ice cream maker to make more logical sense, it does need a big refactor (and TGUIfication) but I'll save that for someone else or a later day.

Replaces the "Ice" button with a "Vanilla" button, because in many ways, vanilla is the default ice cream flavour so having easy access to it makes sense.

Changes the code that colours the ice cream to use an Hue Saturation Value colourspace, and puts some tweaks and clamps on the Saturation and Value (aka brightness) so that the ice creams are vibrant and colourful, not muted and greyish looking. Thanks to @Vi3trice for the code that let me do this.

Updates a whole bunch of the most egregious cases of drinks and a few other reagents that might be commonly used to make ice cream that had nonsensical colour values. (Lemon juice was red. Iced Coffee was blue. Banana juice was red. Etc.)

And finally, deletes a few unused icon states related to ice cream.

## Why It's Good For The Game
This started as just a respriting project for the ice cream thanks to Crumpaloo's sprite work, but the Ice Cream Vat needs some heavy work. I don't have the time for a full refactor, but all these changes should at least make the user end of this machine much more sensical, even if the back end still needs that refactor.

Plus the ice cream looks way better!

## Images of changes
image

Both images contain cones with identical reagents inside them. From top to bottom, they are 30u each of vanilla, tomato, lime, coffee, soy milk, grape juice, banana juice, and gin and sonic.
image

## Testing
Made all sorts of weird ice creams, played around with colourspace clamps and the like for hours.

## Changelog
:cl: PacifistDalek, Crumpaloo, Vi3trice
add: Waffle cones to the ice cream maker
tweak: The ice cream maker makes more vibrant ice cream colours. Replaces Ice with Vanilla as options in the ice cream maker. Updates the colour hex codes assigned to a number of drinks and reagents to make them more realistic
imageadd: New ice cream sprite
/:cl:

### CODE REVIEW

**Comment by DogeDogIs** [comment]
image

I feel that the color of these 3 is very strong, if you can reduce it a little I think it would look better, but keeping the colors vibrant

**Comment by PacifistDalek** [comment]
Tweaked them a touch, it's hard to find a middle ground where it doesn't kill the colour of non-bright ones:
image

(also nothing can help Gin & Sonic it just looks like that no matter what :3 )

File: code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm [hunk 1]
```diff
@@ -141,25 +141,19 @@
...
+	else if(href_list["createchoco"])
 		var/name = generate_name(reagents.get_master_reagent_name())
 		name += " Chocolate Cone"
 		var/obj/item/reagent_containers/food/snacks/icecream/icecreamcup/C
 		C = new/obj/item/reagent_containers/food/snacks/icecream/icecreamcup(loc)
 		C.name = "[name]"
 		C.pixel_x = rand(-8, 8)
 		C.pixel_y = -16
-		reagents.trans_to(C,30)
+		reagents.trans_to(C,50)
```
Comment (S34NW) [inline code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm:156]:
```suggestion
		reagents.trans_to(C, 50)
```

File: code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm [hunk 2]
```diff
@@ -172,7 +166,20 @@
 		C.name = "[name]"
 		C.pixel_x = rand(-8, 8)
 		C.pixel_y = -16
-		reagents.trans_to(C,15)
+		reagents.trans_to(C,30)
```
Comment (S34NW) [inline code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm:169]:
```suggestion
		reagents.trans_to(C, 30)
```

File: code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm [hunk 3]
```diff
@@ -172,7 +166,20 @@
...
+
+	else if(href_list["createwaffle"])
+		var/name = generate_name(reagents.get_master_reagent_name())
+		name += " Waffle Cone"
+		var/obj/item/reagent_containers/food/snacks/icecream/icecreamwafflecone/C
+		C = new/obj/item/reagent_containers/food/snacks/icecream/icecreamwafflecone(loc)
+		C.name = "[name]"
+		C.pixel_x = rand(-8, 8)
+		C.pixel_y = -16
+		reagents.trans_to(C,20)
```
Comment (S34NW) [inline code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm:182]:
```suggestion
		reagents.trans_to(C, 20)
```

File: code/modules/food_and_drinks/food/foods/desserts.dm [hunk 4]
```diff
@@ -19,7 +19,8 @@
 /obj/item/reagent_containers/food/snacks/icecream/update_overlays()
 	. = ..()
 	var/mutable_appearance/filling = mutable_appearance('icons/obj/kitchen.dmi', "icecream_color")
-	filling.color = mix_color_from_reagents(reagents.reagent_list)
+	var/list/reagent_colors = rgb2num(mix_color_from_reagents(reagents.reagent_list), COLORSPACE_HSV)  //switching to HSV colorspace lets us easily manipulate the saturation and brightness independently
+	filling.color = rgb(reagent_colors[1], ((1.5*reagent_colors[2])-10), (clamp(reagent_colors[3], 85, 100)-10)	, space = COLORSPACE_HSV) //Clamping the brightness keeps us from having greyish ice cream while still alowing for a range of colours
```
Comment (S34NW) [inline code/modules/food_and_drinks/food/foods/desserts.dm:23]:
```suggestion
	filling.color = rgb(reagent_colors[1], ((1.5*reagent_colors[2])-10), (clamp(reagent_colors[3], 85, 100)-10), space = COLORSPACE_HSV) //Clamping the brightness keeps us from having greyish ice cream while still alowing for a range of colours
```

File: code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm [hunk 5]
```diff
@@ -141,25 +141,19 @@
 					else
 						reagents.add_reagent("gin",5)
 			else if(ID == 4)
-				if(reagents.total_volume <= 500 & reagents.total_volume >= 15)
-					reagents.add_reagent("cream",(30 - reagents.total_volume))
-				else if(reagents.total_volume <= 15)
-					reagents.add_reagent("cream",(15 - reagents.total_volume))
+				reagents.add_reagent("cream",5)
```
Comment (S34NW) [inline code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm:144]:
```suggestion
				reagents.add_reagent("cream", 5)
```

File: code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm [hunk 6]
```diff
@@ -141,25 +141,19 @@
...
 			else if(ID == 5)
-				if(reagents.total_volume <= 500 & reagents.total_volume >= 15)
-					reagents.add_reagent("water",(30 - reagents.total_volume))
-				else if(reagents.total_volume <= 15)
-					reagents.add_reagent("water",(15 - reagents.total_volume))
+				reagents.add_reagent("vanilla",5)
```
Comment (S34NW) [inline code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm:146]:
```suggestion
				reagents.add_reagent("vanilla", 5)
```

**Review by S34NW — COMMENTED** [review]
This seems fine apart from some small spacing issues.
RGB clamping/manipulation is always a nightmare

**Comment by S34NW** [comment]
Ah yeah and merge conflict needs resolving

File: code/modules/food_and_drinks/food/foods/desserts.dm (hunk 4 above)
Comment (PacifistDalek) [reply code/modules/food_and_drinks/food/foods/desserts.dm:23]: that's the power of math!

**Comment by PacifistDalek** [comment]
Absolutely Everything Broke, and I'm Having TROUBLE fixing it. Gonna close and start fresh on a new branch so I don't give myself a headache.

**Pull request closed without merging**